### DISCUSSION TITLE
ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile"

### DISCUSSION THREAD

**Discussion by vladIsLove-hub** [pnpm/pnpm#4847] [Q&A]
Hello everyone. 
I am facing this problem and nowhere in the documentation is there a detailed answer to this bug. The bottom line is that when we push changes to CI we generally don't need to update pnpm-lock.yml , because it might lead to unexpected consequences in your team. We want everything to be set relative to the lock file. The documentation says that you can use --frozen-lockfile for this, but when you try to use it, this error occurs. And a solution is suggested to disable it, but I don't want to update my lock file as I wrote, it can lead to very unexpected behavior. Please tell me someone was able to solve this problem?

```Scope: all 12 workspace projects
Lockfile is up-to-date, resolution step is skipped
 ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up-to-date with package.json

Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile```

**Comment by jason-ha** [comment]
I am not sure what you mean by relative to the lock file. The `pnpm-lock.yml` file is basically a complete dependency list and map from some attempt to install dependencies. (There can be multiple valid installations from a single dependency set.) Encoded in the lock file is also the list of dependencies that were listed in `package.json` (or the full set from workspaces in mono-repo) so that `pnpm` can know that using the resolved list and map would be valid for the current `package.json`(s). If you change any dependency in a `package.json` file, then you must update the lock file to able to use its content. If you don't want to use the lock file, then you can exclude it from your repository. You may also still need to use `--no-frozen-lockfile` in a CI environment as the output note mentions.
In general lock file use is recommended as it states exactly what dependencies are in use.

**Reply by edaojs** [reply]
the option `--no-frozen-lockfile` is not on documentation, maybe it is a wrong output message.
the option `--frozen-lockfile=false` that is on documentation, didn't work for me
what works for me is  `pnpm install --lockfile=false` which is the same instruction you can set on your .npmrc file.
I hope it helps

**Comment by Daniel-Khodabakhsh** [comment]
The issue is that your local version of `pnpm` is different than the one being used in the pipeline, hence the lockfile being "outdated" → it was generated with an older version of `pnpm`.

The message is not very clear. Ran into the same issue and happened to guess the solution correctly but the wording led me in the right direction.

You can fix this by either locking down the version of `pnpm` used in the pipeline, or make sure to periodically update the version of `pnpm` you use locally. I would suggest the latter to increase your chance of keeping up-to-date with security patches. Executing `npm install pnpm --global` will upgrade your version of `pnpm`.

**Reply by Daniel-Khodabakhsh** [reply]
I created a PR to try to make the message more explicit: https://github.com/pnpm/pnpm/pull/6304
